### PR TITLE
test: Add comprehensive unit tests and CI workflow

### LINKED ISSUE #64: 补充单元测试
当前项目很大的问题是没有任何单元测试内容；

请你分析关键实现，补充测试内容；

然后修改github工作流文件，在ci处运行单元测试，确保通过

### PR DESCRIPTION
**Pull request by minorcell**
Requested by @minorcell

Resolves #64

## Summary
Added comprehensive unit test coverage for the Pick Quote extension and integrated automated testing into the CI pipeline.

### Test Coverage
- **Utils Module** (20 tests)
  - SHA-256 hash generation with edge cases
  - Item hash computation with deduplication logic
  - URL formatting and truncation

- **Database Module** (26 tests)
  - Item CRUD operations (add, update, delete, export)
  - Search and filtering functionality
  - Category management
  - Deduplication logic verification
  - IndexedDB transaction handling

- **Export Module** (13 tests)
  - ZIP file generation with markdown content
  - Image extraction from data URLs
  - Multiple item type handling
  - Edge cases (empty arrays, missing data)

### Infrastructure
- ✅ Jest testing framework with TypeScript support
- ✅ fake-indexeddb for database mocking
- ✅ Polyfills for Node.js environment (TextEncoder, crypto.subtle, structuredClone)
- ✅ GitHub Actions workflow for automated CI testing
- ✅ Test coverage reporting

### Test Results
```
Test Suites: 3 passed, 3 total
Tests:       59 passed, 59 total
Snapshots:   0 total
Time:        5.02 s
```

## Test plan
- [x] All 59 tests pass locally
- [x] CI workflow configured to run on push and PR
- [x] Tests cover critical functionality: hashing, database operations, export features
- [x] Edge cases and error handling tested

Generated with [codeagent](https://github.com/qbox/codeagent)
Co-authored-by: minorcell <120795714+minorcell@users.noreply.github.com>

### CODE REVIEW

File: src/test/setup.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { TextEncoder, TextDecoder } from 'util'
+import { webcrypto } from 'crypto'
+
+// Polyfill TextEncoder/TextDecoder
+global.TextEncoder = TextEncoder
+global.TextDecoder = TextDecoder as any
+
+// Polyfill structuredClone for Node < 17
+if (typeof global.structuredClone === 'undefined') {
+  global.structuredClone = (obj: any) => JSON.parse(JSON.stringify(obj))
```
Comment (minorcell) [inline src/test/setup.ts]: **Security Issue: Unsafe structuredClone polyfill**

This polyfill has several critical issues:
- Cannot handle circular references (will throw)
- Loses functions, Symbols, and other non-JSON-serializable values
- Doesn't match the real `structuredClone` behavior

This could mask production bugs that only appear in Node 17+. Consider using a proper polyfill library like `core-js-pure` or at minimum add validation:

```typescript
if (typeof global.structuredClone === 'undefined') {
  global.structuredClone = (value: any) => {
    try {
      return JSON.parse(JSON.stringify(value))
    } catch (e) {
      throw new Error(`structuredClone polyfill failed: ${e.message}`)
    }
  }
}
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Run tests
+        run: pnpm test
```
Comment (minorcell) [inline .github/workflows/test.yml]: **Performance Issue: Redundant test execution**

This workflow runs tests twice:
1. Line 49: `pnpm test` 
2. Line 52: `pnpm test:coverage`

The second run repeats all tests with coverage instrumentation. Consider running tests only once:

```yaml
- name: Run tests with coverage
  run: pnpm test:coverage
  if: matrix.node-version == '20.x'

- name: Run tests (no coverage)
  run: pnpm test
  if: matrix.node-version == '18.x'
```

This would reduce CI time by ~50% for the Node 20.x job.

File: src/database/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        id: 'text1',
+        type: 'text',
+        content: 'Hello world',
+        source: { title: 'Page 1', url: 'https://example.com/1', site: 'example.com' },
+        createdAt: 1000
+      }))
+      await addItem(createTestItem({
+        id: 'image1',
+        type: 'image',
+        content: 'data:image/png;base64,xyz',
```
Comment (minorcell) [inline src/database/index.test.ts]: **Security Issue: Invalid base64 test data**

The value `'xyz'` is not valid base64. While this test doesn't exercise the export path, having invalid data could mask bugs. If this ever reaches `dataUrlToBlob()` in `zipExport.ts`, it would silently produce corrupted data.

Use a valid base64 image:

```typescript
const validBase64 = 'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg=='
content: `data:image/png;base64,${validBase64}`,
```

**Comment by minorcell** [comment]
### Working on fixing pnpm lockfile sync issue 

**Todo List:**
- [x] Verify current branch and git status
- [ ] Update pnpm-lock.yaml by running `pnpm install`
- [ ] Commit and push the lockfile changes
- [ ] Verify the fix resolves the CI issue

**Progress:**
- ✅ Confirmed on branch: `xgopilot/claude/issue-64-1763533215`
- ✅ Identified new dependencies added: @testing-library/jest-dom, @testing-library/react, @types/jest, fake-indexeddb, jest, jest-environment-jsdom, ts-jest
- 🔄 Running `pnpm install` to update lockfile...

[View job run](http://localhost:5173/ailog?org=minorcell&repo=pick-quote&prId=65&commentId=3551067755)
